lib/csv/csv.nit

```diff
@@ -15,119 +15,156 @@
 # CSV document handling.
 module csv
 
-# Specifies a CSV format.
-class CsvFormat
+redef class Text
+	# Escape the content of `self` for inclusion in a CSV document
+	private fun escape_to_csv(sep_char, delim_char: Char, eol: String): String do
+		var add_sp = to_escape_csv(sep_char, delim_char, eol)
+		if add_sp == 0 then return to_s
+		var bf = new Buffer.with_cap(add_sp + bytelen)
+		bf.add '"'
+		for i in [0 .. length[ do
+			var c = self[i]
+			if c == delim_char then
+				bf.add c
+			end
+			bf.add c
+		end
+		bf.add '"'
+		return bf.to_s
+	end
+
+	# How many more bytes should be allocated for CSV escaping ?
+	private fun to_escape_csv(sep_char, delim_char: Char, eol: String): Int do
```

Comment:
The name doesn't fit the service and it is strangely close to `escape_to_csv`. It could be `escape_extra_length`...